File: module2/src/phuong/case_study/furama_manager/model/person/Customer.java
package phuong.case_study.furama_manager.model.person;

import java.sql.Date;

public class Customer extends Person {
    private String customerRank;
    private String address;

    public Customer() {
    }

    public Customer(String customerRank, String address) {
        this.customerRank = customerRank;
        this.address = address;
    }

    public Customer(String id, String name, Date date, String gender, int idCard, int phoneNumber, String mail, String customerRank, String address) {
        super(id, name, date, gender, idCard, phoneNumber, mail);
        this.customerRank = customerRank;
        this.address = address;
    }

    public String getCustomerRank() {
        return customerRank;
    }

    public void setCustomerRank(String customerRank) {
        this.customerRank = customerRank;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    @Override
    public String toString() {
        return "Customer {" +
                super.toString() +
                ", customerRank = '" + customerRank + '\'' +
                ", address = '" + address + '\'' +
                '}';
    }
}

File: module5/angular-test/src/app/furama-management/model/customer.ts
export class Customer {
  private _id?: number;
  private _name?: string;
  private _age?: number;

  constructor(id: number, name: string, age: number) {
    this._id = id;
    this._name = name;
    this._age = age;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get age(): number {
    return this._age;
  }

  set age(value: number) {
    this._age = value;
  }
}
